benchmarks/scripts/benchmark_field_last.jl

```diff
@@ -0,0 +1,333 @@
+│ funcs                                                               │ time per call (CPU)               │ bw %     │ achieved bw │ n-reads/writes │ n-reps │
+├─────────────────────────────────────────────────────────────────────┼───────────────────────────────────┼──────────┼─────────────┼────────────────┼────────┤
+│     FLD.aos_cart_offset!(X_aos_ref, Y_aos_ref, us; bm, nreps = 100) │ 16 milliseconds, 494 microseconds │ 0.241171 │ 4.91747     │ 4              │ 100    │
+│     FLD.aos_lin_offset!(X_aos, Y_aos, us; bm, nreps = 100)          │ 783 microseconds, 256 nanoseconds │ 5.07871  │ 103.555     │ 4              │ 100    │
+│     FLD.soa_linear_index!(X_soa, Y_soa, us; bm, nreps = 100)        │ 790 microseconds, 894 nanoseconds │ 5.02966  │ 102.555     │ 4              │ 100    │
+│     FLD.soa_cart_index!(X_soa, Y_soa, us; bm, nreps = 100)          │ 12 milliseconds, 522 microseconds │ 0.317663 │ 6.47714     │ 4              │ 100    │
+└─────────────────────────────────────────────────────────────────────┴───────────────────────────────────┴──────────┴─────────────┴────────────────┴────────┘
+```
+
+=#
+
+#! format: off
+module BenchmarkFieldLastIndex
+
+include("benchmark_utils.jl")
+
+@inline function const_linear_index(us::UniversalSizesStatic, I, field_index)
+    n = (get_Nv(us), get_Nij(us), get_Nij(us), get_Nh(us), 1)
+    i = I + prod(n)*field_index
+    return i
+end
+
+@inline function const_linear_index_reference(us::UniversalSizesStatic, I, field_index)
+    CI = CartesianIndices((get_Nv(us), get_Nij(us), get_Nij(us), get_Nh(us), 1))
+    LI = LinearIndices((get_Nv(us), get_Nij(us), get_Nij(us), get_Nh(us), field_index+1))
+    return LI[CI[I] + CartesianIndex((0, 0, 0, 0, field_index))]
+end
+
+# add3(x1, x2, x3) = x1 + x2 + x3
+add3(x1, x2, x3) = x1
+
+function aos_cart_offset!(X, Y, us; nreps = 1, bm=nothing, n_trials = 30)
+    if Y isa Array
+        e = Inf
+        CI = CartesianIndices((get_Nv(us), get_Nij(us), get_Nij(us), get_Nh(us), 1))
+        for t in 1:n_trials
+            et = Base.@elapsed begin
+                for i in 1:nreps
+                    @inbounds @simd for I in 1:get_N(us)
+                        CI1 = CI[I]
+                        CI2 = CI1 + CartesianIndex((0, 0, 0, 0, 1))
+                        CI3 = CI1 + CartesianIndex((0, 0, 0, 0, 2))
+                        Y[CI1] = add3(X[CI1], X[CI2], X[CI3])
+                    end
+                end
+            end
+            e = min(e, et)
+        end
+    else
+        e = Inf
+        kernel = CUDA.@cuda always_inline = true launch = false aos_cart_offset_kernel!(X,Y,us)
+        config = CUDA.launch_configuration(kernel.fun)
+        threads = min(get_N(us), config.threads)
+        blocks = cld(get_N(us), threads)
+        for t in 1:n_trials
+            et = CUDA.@elapsed begin
```

Comment:
Should this timer be moved outside the encompassing `add` function? 
Can we use benchmark tools for gathering timing information?

Reply:
No, this was done to avoid measuring launch latency. From experimentation, I found that it turned out to not be important, but I did this to be careful.